File: 001_userInput/001_easy.clj
(defmacro interrogate [& queries]
  `(for [query# [~@queries]]
     (do
       (println query#)
       (read-line))))

(let [[n, a, rn] (interrogate "What is your name?"
                              "How old are you?"
                              "What is your reddit username?")]
  (println (str "your name is " n ", you are " a
                " years old, and your username is " rn)))

File: 001_easy_userInput/001_easy.hs
import System.IO

main = do
    putStrLn "What is your name?"
    name <- getLine
    putStrLn "How many years have you been alive?"
    age <- getLine
    putStrLn "What is your reddit username?"
    username <- getLine
    let msg = "your name is " ++ name ++ ", you are " ++ age ++ " years old, and your reddit username is " ++ username
        filename = "userinfo.txt"
    putStrLn $ "writing the following message to " ++ filename
    writeFile filename msg -- extra credit
    putStrLn $ msg

